watertap/ui/fsapi.py

```diff
@@ -66,21 +66,39 @@ def __init__(
 
 
 class ModelExport(BaseModel):
-    """A variable, expression, or parameter."""
+    """A variable, expression, or parameter.
+
+    The object may be passed as a valid Pyomo model component with the 'obj' attribute
+    or, if it may not exist yet in the flowsheet until after solving the model, as a
+    string with the 'deferred_obj'. An example of using deferred_obj is shown below::
+
+        exports.add(
+            deferred_obj="fs.leach.solid_outlet.flow_mass[0]",
+            name="solid flow mass",
+            rounding=4,
+            ui_units=pyo.units.kg/pyo.units.hour,
+            display_units="kg/hr",
+            description="solid flow mass",
+            is_input=False,
+            is_output=True,
+            output_category="solids"
+        )
+
+    Note that it makes no sense to use both attributes at the same time.
```

Comment:
This is more of a high-level comment, but I'm wondering if making the `obj` field more versatile (i.e. `obj: Union[_SupportedObjType, str]`) could be an alternative to the current solution of having separate fields and having to enforce (or trust the clients to comply with) their mutual-exclusiveness.

However, I don't know enough about the concrete implementation details so this suggestion might not even make sense (e.g. making it impossible to undefer `str` objects in the very scenarios that require deferred objects in the first place).

Reply:
I think the current solution is more explicit

Reply:
on second thought, sure..